recipes/pcs/hpc_ready_ami/assets/scripts/optimize-performance.sh

```diff
@@ -48,19 +52,22 @@ disable_deeper_cstates() {
 
         logger "Disabling deeper C-States" "INFO"
 
-        GRUB_COMMAND="grub2-mkconfig -o /boot/grub2/grub.cfg"
-        GRUB_SUB='s/^GRUB_CMDLINE_LINUX_DEFAULT=".*"/GRUB_CMDLINE_LINUX_DEFAULT="intel_idle.max_cstate=0 processor.max_cstate=1"/'
-
-        if [ "${OS}" == "ubuntu" ]; then
-            GRUB_COMMAND="/usr/sbin/update-grub"
-            GRUB_SUB='s/^GRUB_CMDLINE_LINUX=".*"/GRUB_CMDLINE_LINUX="intel_idle.max_cstate=0 processor.max_cstate=1"/'
-        fi
-
-        sudo sed -i "${GRUB_SUB}" /etc/default/grub
-        sudo ${GRUB_COMMAND}
-
+        # See https://github.com/aws/aws-parallelcluster-cookbook/commit/1f5911817d10b9ca00a706c94d50ef5e756b81e7
+        case "${OS}-${VERSION}" in
```

Comment:
Would this be better handled by checking if grubby exists? That should work for other distros like Alma as well

```sh
if ! command -v grubby 2>&1 >/dev/null
```

Reply:
Good point. However, the Pcluster devs claim `grubby` does not take effect on "some of the old OSs"  https://github.com/aws/aws-parallelcluster-cookbook/commit/1f5911817d10b9ca00a706c94d50ef5e756b81e7

Since we only support ubuntu,RHEL,rocky,Amzn at the moment I would like to follow the Pcluster best practices.